ownCloud/Static Login/Interface/StaticLoginSetupViewController.swift

```diff
@@ -170,6 +223,79 @@ class StaticLoginSetupViewController : StaticLoginStepViewController {
 		return connection
 	}
 
+	@objc func helpAction() {
+		if let helpURL = self.profile.helpURL {
+			let alert = ThemedAlertController(title: "Do you want to open the following URL?".localized,
+							  message: helpURL.absoluteString,
+							  preferredStyle: .alert)
+
+			let okAction = UIAlertAction(title: "OK", style: .default) { (_) in
+				UIApplication.shared.open(helpURL, options: [:], completionHandler: nil)
+			}
+			let cancelAction = UIAlertAction(title: "Cancel".localized, style: .cancel)
+			alert.addAction(okAction)
+			alert.addAction(cancelAction)
+			self.present(alert, animated: true)
+		}
+	}
+
+	@objc func proceedWithURL() {
+		var error = false
+		if let value = self.urlString, value.count > 0 {
+			if let normalizedURL = NSURL(username: nil, password: nil, afterNormalizingURLString: value, protocolWasPrepended: nil) {
+				if let allowedHosts = self.profile.allowedHosts, allowedHosts.count > 0, let host = normalizedURL.host {
+					if !allowedHosts.contains(where: { (allowedHost) -> Bool in
+						return host.hasSuffix(allowedHost) ? true : false
+					}) {
+						error = true
+					}
+				}
+
+				if !error {
+					self.bookmark = OCBookmark(for: normalizedURL as URL)
+					self.proceedWithLogin()
+				}
+			} else {
+				error = true
+			}
+		} else {
+			error = true
+		}
+
+		if error {
+			let alert = ThemedAlertController(title: "Wrong URL".localized,
+							  message: "Please enter an valid URL".localized,
```

Comment:
Little typo here. That should be `Please enter a valid URL`.

Reply:
fixed